text/0000-macro-lifetimes.md

```diff
@@ -0,0 +1,51 @@
+- Feature Name: Allow `lifetime` specifiers to be passed to macros
+- Start Date: 2016-04-22
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `lifetime` specifier for `macro_rules!` patterns, that matches any valid
+lifetime.
+
+# Motivation
+[motivation]: #motivation
+
+Certain classes of macros are completely impossible without the ability to pass
+lifetimes. Specifically, anything that wants to implement a trait from inside of
+a macro is going to need to deal with lifetimes eventually. They're also
+commonly needed for any macros that need to deal with types in a more granular
+way than just `ty`.
+
+Since a lifetime is a single token, the only way to match against a lifetime is
+by capturing it as `tt`. Something like `'$lifetime:ident` would fail to
+compile. This is extremely limiting, as it becomes difficult to sanitize input,
+and `tt` is extremely difficult to use in a sequence without using awkward
+separators.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC proposes adding `lifetime` as an additional specifier to
+`macro_rules!` (alternatively: `life` or `lt`). Since a lifetime acts very much
```

Comment:
I agree that the long word `lifetime` seems good. I don't see a reason to save letters here.